node_modules/gh-series/tests/test-patterns.js

```diff
@@ -0,0 +1,259 @@
+            // Create a serie with 3 events that can aggregate
+            setupAggregation(function(eventA, eventB, eventC, serie) {
+
+                // Assert all the metadata is present
+                assert.ok(serie.metadata);
+                assert.ok(serie.metadata.locations);
+                assert.ok(serie.metadata.organisers);
+                assert.ok(serie.metadata.pattern);
+
+                // Assert the metadata is aggregated and returned in the correct order
+                assert.deepEqual(serie.metadata.locations, ['Room 1', 'Room 2']);
+                assert.deepEqual(serie.metadata.organisers, ['Dr Jack Daniels', 'Prof William Lawsons']);
+                assert.strictEqual(serie.metadata.pattern, 'Mi3-5 Tu 13');
+                return callback();
+            });
+        });
+
+        /**
+         * Test that verifies that serie metadata is updated when events their locations change
+         */
+        it('verify serie metadata is updated when events their locations change', function(callback) {
+            // Create a serie with 3 events that can aggregate
+            setupAggregation(function(eventA, eventB, eventC, serie) {
+                // Sanity check the locations aggregated
+                assert.deepEqual(serie.metadata.locations, ['Room 1', 'Room 2']);
+
+                // Update the location of an event
+                EventsTestUtil.assertUpdateEvent(global.tests.admins.cam2014.client, eventA.id, {'location': 'Room 3'}, function() {
+
+                    // Get the new event metadata and verify the new location is now in the metadata
+                    SeriesTestUtil.assertGetSerie(global.tests.admins.cam2014.client, serie.id, null, function(serie) {
+                        assert.deepEqual(serie.metadata.locations, ['Room 1', 'Room 2', 'Room 3']);
+
+                        // All events should take place in the same location
+                        EventsTestUtil.assertUpdateEvent(global.tests.admins.cam2014.client, eventB.id, {'location': 'Room 3'}, function() {
+                            EventsTestUtil.assertUpdateEvent(global.tests.admins.cam2014.client, eventC.id, {'location': 'Room 3'}, function() {
+                                SeriesTestUtil.assertGetSerie(global.tests.admins.cam2014.client, serie.id, null, function(serie) {
+                                    assert.deepEqual(serie.metadata.locations, ['Room 3']);
+                                    return callback();
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that serie metadata is updated when events their organisers change
+         */
+        it('verify serie metadata is updated when events their organisers change', function(callback) {
+            // TODO
```

Comment:
That's going to make the PR more bloated though, I'd like to do this in a follow-up PR, so review can stay "bite-sized"